src/Tribe/Settings.php

```diff
@@ -470,23 +650,144 @@ public function generate_page(): void {
 	}
 
 	/**
-	 * Generate the tabs in the settings screen.
+	 * Generate the tab navigation in the settings screen.
 	 *
-	 * @since TBD
+	 * Each level of the tab nav is a unordered list inside a nav element.
+	 * This function generates the structure and the generate_tab
+	 * function creates the individual list items.
 	 *
-	 * @return void
+	 * @since TBD
 	 */
 	public function generate_tabs(): void {
-		if ( is_array( $this->tabs ) && ! empty( $this->tabs ) ) {
-			echo '<h2 id="tribe-settings-tabs" class="nav-tab-wrapper">';
-			foreach ( $this->tabs as $tab => $name ) {
-				$url   = $this->get_tab_url( $tab );
-				$class = ( $tab == $this->current_tab ) ? ' nav-tab-active' : '';
-				echo '<a id="' . esc_attr( $tab ) . '" class="nav-tab' . esc_attr( $class ) . '" href="' . esc_url( $url ) . '">' . esc_html( $name ) . '</a>';
+		if ( ! is_array( $this->tabs ) || empty( $this->tabs ) ) {
+			return;
+		}
+
+		ob_start();
+		?>
+			<nav id="tribe-settings-tabs" class="tec-nav__wrapper">
+				<ul class="tec-nav">
+					<li class="tec-nav__tab tec-nav__tab--skip-link">
+						<a href="#tec-settings-form" class="screen-reader-shortcut"><?php esc_html_e( 'Skip to tab content', 'tribe-common' ); ?></a>
+					</li>
+				<?php
+
+				foreach ( $this->tabs as $tab ) {
+					if ( ! empty( $tab->parent ) ) {
+						// This tab belongs in the subnav!
+						continue;
+					}
+
+					$this->generate_tab( $tab );
+				}
+				?>
+				</ul>
+				<?php do_action( 'tribe_settings_after_tabs' ); ?>
+			</nav>
+		<?php
+		echo ob_get_clean(); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped,StellarWP.XSS.EscapeOutput.OutputNotEscaped
+	}
+
+	/**
+	 * Generate the content for a single specified tab.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Settings_Tab $tab The tab object.
+	 */
+	public function generate_tab( $tab ) {
+		if ( ! $tab instanceof Tribe__Settings_Tab ) {
+			return;
+		}
+
+		$url   = $this->get_tab_url( $tab->id );
+		$class = [ 'tec-nav__tab' ];
+		$current_tab = $this->get_current_tab();
+
+		if ( $tab->has_children() && isset( $tab->children[ $current_tab ] ) ) {
+			// Current tab is a child tab of passed tab.
+			$class[] = 'tec-nav__tab--subnav-active';
+		} elseif ( $tab->has_children() && $tab->id === $current_tab ) {
+			// Current tab is a parent tab. Set to first child.
+			$this->current_tab = array_key_first( $tab->children );
+			$class[]           = 'tec-nav__tab--subnav-active';
+		} elseif ( $tab->id === $current_tab ) {
+			$class[] = 'tec-nav__tab--active';
+		}
+
+		ob_start();
+		?>
+		<li <?php tribe_classes( $class ); ?>>
+			<a
+				id="<?php echo esc_attr( $tab->id ); ?>"
+				class="tec-nav__link"
+				href="<?php echo esc_url( $url ); ?>"><?php echo esc_html( $tab->name ); ?></a>
+			<?php
+			if ( $tab->has_children() ) {
+				$this->generate_subnav( $tab, $tab->children );
```

Comment:
Two things here:

1. This feels like something the parent tab should be responsible for (generating the subnav content)
2. I'd rather see the tab's children accessed via a method instead of a `public` property.

Reply:
1. I disagree - we set up the tabs here, it seems natural we'd set up the subtabs as well. (renamed function)
2. Agreed - done.